{
  try {
    _log.trace(msg);
  }
 catch (  Exception e) {
    printMsg(msg);
  }
}
